app/models/foreman_tasks/recurring_logic.rb

```diff
@@ -83,6 +83,7 @@ def cancel
 
     def next_occurrence_time(time = Time.zone.now)
       @parser ||= CronParser.new(cron_line, Time.zone)
+      return @parser.next(@parser.last(time)) if @parser.next(@parser.last(time)) >= time && tasks.count == 0
```

Comment:
this logic works and is correct, but is a little hard to follow. What data is `@parser.next(@parser.last(time))` capturing? is it the next run, but backdated one run?

Reply:
So with a start_at 12:46 if we have a cron "46 * * * *" , @parser.next(12:46) returns 13:46 while parser.last returns 11:46 and parser.next(11:46) returns 12:46 which is when we want the task to run..

Reply:
Can you add a comment saying something like "CronParser's `next` is not inclusive of the current time, so we step backwards one run to calculate the next run"? My concern is that if I came across this line in the future, I would not know what it was doing at all.

Reply:
One nitpick, it would be nice if you could avoid generating the time twice. Something like this might work
```ruby
before_next = @parser.next(@parser.last(time))
return before_next if before_next >= time && task.count.zero?
@parser.next(time)
```